R/zchunk_L124.nonco2_unmgd_R_S_T_Y.R

```diff
@@ -43,80 +41,182 @@ module_emissions_L124.nonco2_unmgd_R_S_T_Y_DISABLED <- function(command, ...) {
+      select(IPCC, Non.CO2, sector, iso, GCAM_region_ID, year, value) ->
+      EDGAR_history
+
+    if(OLD_DATA_SYSTEM_BEHAVIOR) {
+      # The old data system processed data in wide format. There was a na.omit() in the code that effectively removed
+      # any region/sector combination where the time series was incomplete. We'd like to keep this information.
+      # Additionally, the old data system never included NH3 in the processing. The emissions are small
+      # but we should keep them.
+      EDGAR_history %>%
+        filter(Non.CO2 != "NH3" ) %>%                                                    # Remove NH3 for consistency with old data
+        filter(year <= 2008) %>%                                                         # Old data didn't care if post-2008 data was missing so remove it here
+        spread(year, value) %>%                                                          # Convert to wide format
+        na.omit() %>%                                                                    # Remove any row with an NA (i.e., incomplete time series)
+        gather(year, value, -GCAM_region_ID, -iso, -sector, -Non.CO2, -IPCC) %>%         # Convert back to long format
+        mutate(year = as.integer( year )) ->                                             # Convert year back to integer form (not sure why this changes type)
+        EDGAR_history
+
+    } else {
+      #   ... new code with a fix
+    }
+
+    # Prepare EDGAR emissions for use (continued)
+    EDGAR_history %>%
+      # Extrapolate using average of last five years --- 2010 is problematic here because of deforestation (i.e., we don't want to rely on 2005 as last year of data)
+      filter(year %in% 2004:2008) %>%
+      group_by(GCAM_region_ID, iso, sector, Non.CO2, IPCC) %>%
+      summarize(value = mean(value)) %>%
+      mutate(year = as.integer(2010)) %>%
+      bind_rows(EDGAR_history) %>%                                                      # Bind extrapolated 2010 data to the rest of the EDGAR data
+      group_by(GCAM_region_ID, sector, Non.CO2, year) %>%
+      summarize(value = sum(value)) %>%                                                 # Aggregate by region, sector, gas, and year
+      na.omit() %>%
+      mutate(value = value * CONV_GG_TG) ->                                             # Convert to Tg
+      EDGAR_history
+
+    # Part 1: Grassland burning
+    # Downscale regional grassland burning emissions to GLU based on the share of land in each GLU
+    L124.LC_bm2_R_Grass_Yh_GLU_adj %>%
+      group_by(GCAM_region_ID, year) %>%
+      mutate(land_share = value / sum(value)) %>%                                                           # Compute the share of regional grassland in each GLU
+      select(-value) %>%
+      left_join(filter(EDGAR_history, sector == "grassland"), by = c("GCAM_region_ID", "year")) %>%         # Map in EDGAR grassland emissions
```

Comment:
Yes, there are NAs at this point. This is because there are regions where we have grassland area but no grassland emissions (e.g., region 3). What I don't know is whether this problem will go away when we turn off OLD_DATA_SYSTEM. For now, I'll put a note.